code/game/machinery/doors/door.dm

```diff
@@ -419,10 +419,10 @@
 
 	do_animate("closing")
 	layer = closingLayer
-	sleep(0.5 SECONDS)
+	SLEEP_NOT_DEL(0.5 SECONDS)
 	set_density(TRUE)
 	flags_1 |= PREVENT_CLICK_UNDER_1
-	sleep(0.5 SECONDS)
+	SLEEP_NOT_DEL(0.5 SECONDS)
```

Comment:
couldn't we use this in `open()` too? it's right above this proc and it involves the same type of airlock sleeps you talk about in this PR.

Reply:
if we're being qdeleted anyways may as well early return out of the work in that proc too

Reply:
yes